src/App.js

```diff
@@ -11,6 +12,20 @@ import ProtectedRoute from './components/ProtectedRoute';
 
 function App() {
   const [isLoggedIn, setIsLoggedIn] = useState(false);
+
+  useEffect(() => {
+    setIsLoggedIn(false);
+    if (Cookies.get('token')) {
+      console.log('token: ' + Cookies.get('token'));
+      setIsLoggedIn(true);
+      // Cookies.remove('token');
+      console.log(isLoggedIn);
+    } else {
+      console.log('no cookie');
+      setIsLoggedIn(false);
+    }
+  }, [setIsLoggedIn]);
```

Comment:
so `useEffect` is triggered on the initial render, and only runs again when anything in the second argument changes.

Here you are passing in `[setIsLoggedIn]` which is a function, and it's value will not update. You can remove that and replace it with an empty array

I'm guessing that the reason that setting the cookie in the login function is not triggering this is because this component is unaware that anything changes, and has no reason to re-trigger the `useEffect` hook